packages/synapse-sdk/src/telemetry/shutdown-utils.ts

```diff
@@ -0,0 +1,54 @@
+// telemetry/shutdown.ts
+import { removeGlobalFetchWrapper } from './fetch-wrapper.ts'
+import type { Sentry as SentryType } from './get-sentry.ts'
+
+export function setupShutdownHooks(sentry: SentryType, opts: { timeoutMs?: number } = {}) {
+  const g = globalThis as any
+  const timeout = opts.timeoutMs ?? 2000
+  let shuttingDown = false
+
+  // NOTE: sentry handles uncaughtException and unhandledRejection. see https://docs.sentry.io/platforms/javascript/troubleshooting/#third-party-promise-libraries
+
+  if (typeof g.window !== 'undefined' && typeof g.document !== 'undefined') {
```

Comment:
I assume we have a more shorthand / self-documenting way at this point for knowing whether we're in a browser or not right?  Basically, we already have the right type of Sentry instance so we should already know whether we're handling a browser a not without having to these less readable checks.